### PR TITLE
Improve Documentation to correctly state what repositories to get for various Linux distros

### PR DESCRIPTION
**Pull request by iamarchit123**
This is to improvement on documentation to help new users get correct repositories to install and use mlpack

### CODE REVIEW

File: doc/guide/build.hpp [hunk 1]
```diff
@@ -8,6 +8,18 @@ FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
 and so it may be easier to use the package manager for your system.  For example, 
 on Ubuntu, you can install mlpack with the following command:
 
+@code
```
Comment (zoq) [inline doc/guide/build.hpp:11]: I can't directly comment on the line above, anyway, I think it makes sense to be more verbose here, since mlpack-bin will only install the mlpack executables. So I would rephrase it to:

For example, on Ubuntu, you can install the mlpack executables (e.g. mlpack_pca, mlpack_kmeans) with the following command:

File: doc/guide/build.hpp [hunk 2]
```diff
@@ -8,6 +8,18 @@ FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
...
+$ sudo apt-get install mlpack-bin
+@endcode
+
+On Fedora or Red Hat:
+
+@code
+$ sudo dnf install mlpack
```
Comment (zoq) [inline doc/guide/build.hpp:18]: I'm not sure what this package entails, lib + executables?

File: doc/guide/build.hpp [hunk 3]
```diff
@@ -8,6 +8,18 @@ FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
...
+@endcode
+
+For installing only header files for development purposes one could use:
```
Comment (zoq) [inline doc/guide/build.hpp:21]: `libmlpack-dev` installs the development libs, mlpack is not a header only library at least not yet.

**Review by zoq — COMMENTED** [review]
Thanks for putting this together, I left some hopefully helpful comments.

File: doc/guide/build.hpp (hunk 3 above)
Comment (iamarchit123) [reply doc/guide/build.hpp:21]: @zoq I got this info from here https://www.mlpack.org/doc/stable/doxygen/cli_quickstart.html .But looks like mlpack packages don't exist on fedora. I did dnf list all on a fedora instance and found that there is no such package that matches mlpack string on Fedora repository.If it exists by alternate name please suggest one and I will change it. Else i will remove this line as mlpack may not exist on fedora/red hat repositories

Comment (iamarchit123) [reply doc/guide/build.hpp:21]: I checked on Red Hat 8.3.1-5 dont know if earlier versions support.

File: doc/guide/build.hpp (hunk 2 above)
Comment (iamarchit123) [reply doc/guide/build.hpp:18]: @zoq  I got this info from here https://www.mlpack.org/doc/stable/doxygen/cli_quickstart.html .But looks like mlpack packages don't exist on fedora. I did dnf list all on a fedora instance and found that there is no such package that matches mlpack string on Fedora repository.If it exists by alternate name please suggest one and I will change it. Else i will remove this line as mlpack may not exist on fedora/red hat repositories

Comment (iamarchit123) [reply doc/guide/build.hpp:18]: I checked on Red Hat 8.3.1-5 dont know if earlier versions support.

File: doc/guide/build.hpp (hunk 1 above)
Comment (iamarchit123) [reply doc/guide/build.hpp:11]: Yes sure will change this

File: doc/guide/build.hpp (hunk 3 above)
Comment (iamarchit123) [reply doc/guide/build.hpp:21]: Yes got it. Earlier i was noot seeing the libmlpack.so file will add the info here.

File: doc/guide/build.hpp (hunk 2 above)
Comment (zoq) [reply doc/guide/build.hpp:18]: @rcurtin is the maintainer of the package, so pretty sure he knows.

File: README.md [hunk 4]
```diff
@@ -148,7 +148,10 @@ FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
 and so it may be easier to use the package manager for your system.  For example, 
 on Ubuntu, you can install mlpack with the following command:
 
-    $ sudo apt-get install libmlpack-dev
+    $ sudo apt-get install mlpack-bin
```
Comment (rcurtin) [inline README.md:151]: We probably want users to install headers too, so maybe `sudo apt-get install libmlpack-dev mlpack-bin`?  We could also change the paragraph above to read `you can install the mlpack header files and command line executables with the following command:`.

File: README.md [hunk 5]
```diff
@@ -148,7 +148,10 @@ FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
...
+
+On Fedora or Red Hat:
+    $ sudo dnf install mlpack
```
Comment (rcurtin) [inline README.md:154]: We might need `mlpack-devel` and `mlpack-bin` too to match the Ubuntu command. `mlpack-bin` depends on `mlpack`, so we can just have `mlpack-bin`.

File: doc/guide/build.hpp (hunk 2 above)
Comment (rcurtin) [reply doc/guide/build.hpp:18]: This is just the library.  `mlpack-bin` will contain the command-line programs.

**Review by rcurtin — COMMENTED** [review]
Hey @iamarchit123, thanks for working on this! I think it is an improvement.  Just some simple comments---let me know if I can clarify any of them. :+1:

File: doc/guide/build.hpp (hunk 2 above)
Comment (iamarchit123) [reply doc/guide/build.hpp:18]: But @rcurtin is mlpack their in ;latest red hat repositories. It's not their in Red hat 8.3.1 I believe.

File: README.md (hunk 4 above)
Comment (iamarchit123) [reply README.md:151]: Ya that makes sense. Ill change this and upload new patch.

File: README.md (hunk 5 above)
Comment (iamarchit123) [reply README.md:154]: But @rcurtin are mlpack or related headers and binaries their in latest red hat repositories. It's not their in Red hat 8.3.1 I believe.

Comment (rcurtin) [reply README.md:154]: Yeah, for RHEL it's in the EPEL repositories.  You could just write `In Fedora or Red Hat (EPEL):`.

**Comment by iamarchit123** [comment]
@rcurtin @zoq can you help review this again. I incorporated your changes as well.

File: doc/guide/build.hpp [hunk 6]
```diff
@@ -6,7 +6,19 @@ This document discusses how to build mlpack from source. These build directions
 will work for any Linux-like shell environment (for example Ubuntu, macOS,
 FreeBSD etc). However, mlpack is in the repositories of many Linux distributions
 and so it may be easier to use the package manager for your system.  For example,
-on Ubuntu, you can install mlpack with the following command:
+on Ubuntu, you can install mlpack library and command line executables(eg. mlpack_pca, mlpack_kmeans etc.) with the following command:
```
Comment (shrit) [inline doc/guide/build.hpp:9]:
```suggestion
on Ubuntu, you can install mlpack library and command-line executables (eg. mlpack_pca, mlpack_kmeans, etc.) with the following command:
```

**Comment by iamarchit123** [comment]
THanks @shrit  updated in README.md also

File: README.md [hunk 7]
```diff
@@ -146,9 +146,12 @@ This document discusses how to build mlpack from source. These build directions
 will work for any Linux-like shell environment (for example Ubuntu, macOS,
 FreeBSD etc). However, mlpack is in the repositories of many Linux distributions 
 and so it may be easier to use the package manager for your system.  For example, 
-on Ubuntu, you can install mlpack with the following command:
+on Ubuntu, you can install mlpack library and command-line executbles(eg. mlpack_pca, mlpack_kmeans etc.) with the following command:
```
Comment (rcurtin) [inline README.md:149]: Can you wrap this line since it is longer than 80 characters please?  :)

Also it should read `the mlpack library`, `eg.` should be `e.g.`, and `executbles` should be `executables`.  (Sorry for being pedantic.)

These comments also apply to the other place you've made this change in `build.hpp`. :+1:

File: doc/guide/build.hpp [hunk 8]
```diff
@@ -6,7 +6,19 @@ This document discusses how to build mlpack from source. These build directions
...
+$ sudo apt-get install libmlpack-dev mlpack-bin
+@endcode
+
+On Fedora or Red Hat(EPEL):
+
+@code
+$ sudo dnf install mlpack-devel mlpack-bin
+@endcode
+
+For installing only header files and lib for development purposes one could use:
```
Comment (rcurtin) [inline doc/guide/build.hpp:21]:
```suggestion
For installing only the header files and library for building C++ applications on
top of mlpack, one could use:
```
Just a simple suggestion for clarification.  Otherwise we might imply that one can work on the mlpack source code this way, which is not true.

**Review by rcurtin — COMMENTED** [review]
Hey @iamarchit123, looking good!  Just two more little comments. :+1:

File: README.md (hunk 7 above)
Comment (iamarchit123) [reply README.md:149]: No ..no its fine thanks for the change sir I come earlier from linux kernel background(i work in my company as linux developer) and I don't know if you have run into any of the maintainers of linux kernel they will easily blast you openly in front of everyone for minor details. I am suprised how I was not careful to check all this. Sorry for the  inconvenience let me correct it.

**Comment by iamarchit123** [comment]
@rcurtin  took care of your comments

**Review by zoq — APPROVED** [review]
Looks good to me, thanks for keeping up with the comments.

**Review by rcurtin — APPROVED** [review]
Thanks @iamarchit123!
